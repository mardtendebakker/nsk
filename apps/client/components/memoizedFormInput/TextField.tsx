import React, { memo } from 'react';
import { TextField as BaseTextField } from '@mui/material';

function TextField(props) {
  // eslint-disable-next-line react/jsx-props-no-spreading
  return <BaseTextField fullWidth variant="outlined" {...props} />;
}

export default memo(
  TextField,
  (prevProps, nextProps) => JSON.stringify(prevProps) === JSON.stringify(nextProps),
);
