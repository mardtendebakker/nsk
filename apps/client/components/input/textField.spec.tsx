import { render } from '@testing-library/react';
import { TextField as BaseTextField, TextFieldProps } from '@mui/material';
import TextField from './textField';

jest.mock('@mui/material/TextField', () => jest.fn(() => null));

describe('TextField', () => {
  const props: TextFieldProps = {
    InputLabelProps: {
      shrink: true,
      sx: {
        mb: '.5rem',
        position: 'relative',
        transform: 'unset',
      },
    },
    error: true,
    helperText: 'helper',
    label: 'label',
    name: 'name',
    onChange: () => {},
    placeholder: 'placeholder',
    size: 'medium',
    sx: {
      bgcolor: 'red',
      legend: { width: 0 },
    },
    value: 'value',
    about: 'about',
  };

  it('matches snapshot', () => {
    const { asFragment } = render(<TextField {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('passes props to BaseTextField', () => {
    render(<TextField {...props} />);

    expect(BaseTextField).toBeCalledWith(props, {});
  });
});
