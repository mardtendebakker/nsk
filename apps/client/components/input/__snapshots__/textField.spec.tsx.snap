// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField matches snapshot 1`] = `<DocumentFragment />`;
